implementations/rust/ockam/ockam_api/src/cloud/email_address.rs

```diff
@@ -0,0 +1,210 @@
+        EmailAddress::parse(value)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use quickcheck::{quickcheck, Arbitrary, Gen, TestResult};
+
+    quickcheck! {
+        /// Valid addresses should be parsed ok
+        fn parse_valid_address(email_address: ValidEmailAddress) -> TestResult {
+            match EmailAddress::parse(&email_address.0.0) {
+                Ok(_) => TestResult::passed(),
+                Err(e) => TestResult::error(format!("{e:?}")),
+            }
+        }
+
+        /// Invalid addresses should fail when parsed
+        fn parse_invalid_address(email_address: InvalidEmailAddress) -> TestResult {
+            match EmailAddress::parse(&email_address.0.0) {
+                Ok(_) => TestResult::error(format!("the email address {} should not be valid", email_address.0)),
+                Err(_) => TestResult::passed(),
+            }
+        }
+
+        /// Check that lowercase comparison is done to determine if 2 email addresses are equal
+        fn email_equality(email_address1: EqualEmailAddress, email_address2: EqualEmailAddress) -> bool {
+            email_address1.0 == email_address2.0
+        }
+    }
+
+    /// HELPERS
+
+    /// This newtype generates equal email addresses
+    #[derive(Clone, PartialEq, Eq, Debug)]
+    struct EqualEmailAddress(EmailAddress);
+
+    impl Arbitrary for EqualEmailAddress {
+        fn arbitrary(gen: &mut Gen) -> Self {
+            EqualEmailAddress(EmailAddress(
+                gen.choose(&[
+                    "test@ockam.io",
+                    "test@ockam.io",
+                    "tEst@ockam.io",
+                    "test@Ockam.io",
+                    "TEST@OCKAM.IO",
+                ])
+                .unwrap()
+                .to_string(),
+            ))
+        }
+    }
+
+    /// This newtype generates valid email addresses according to the HTML5 regex
+    /// The examples are taken from: https://en.wikipedia.org/wiki/Email_address#Valid_email_addresses
+    /// and commented out examples are examples which cannot be successfully parsed with the regex
+    #[derive(Clone, PartialEq, Eq, Debug)]
+    struct ValidEmailAddress(EmailAddress);
+
+    impl Arbitrary for ValidEmailAddress {
+        fn arbitrary(gen: &mut Gen) -> Self {
+            ValidEmailAddress(EmailAddress(
+                gen.choose(&[
+                    "simple@example.com",
+                    "very.common@example.com",
+                    "x@example.com",
+                    "long.email-address-with-hyphens@and.subdomains.example.com",
+                    "user.name+tag+sorting@example.com",
+                    "name/surname@example.com",
+                    "admin@example",
+                    "example@s.example",
+                    "mailhost!username@example.org",
+                    "user%example.com@example.org",
+                    "user-@example.org",
+                    // "\" \"@example.org",
```

Comment:
Did you intentionally leave this?

Reply:
Yes, I explain this in the comment [here](https://github.com/build-trust/ockam/pull/7205/files/7309b9834914ff3cd210f279f076040eb560abc2#diff-c85e5db2e533f3560c0ea50cb110aeb59ba0ac50b08e8e134347264d66c255e3R142). I'm going to bring that comment closer.